test/e2e/networkpolicy_test.go

```diff
@@ -31,24 +31,34 @@ import (
 )
 
 func TestDifferentNamedPorts(t *testing.T) {
+	// TODO: add check for IPv6 address after NetworkPolicy is supported in IPv6
+	skipIfIPv6Cluster(t)
+	skipIfNotIPv4Cluster(t)
 	data, err := setupTest(t)
 	if err != nil {
 		t.Fatalf("Error when setting up test: %v", err)
 	}
 	defer teardownTest(t, data)
 
 	server0Port := 80
-	_, server0IP, cleanupFunc := createAndWaitForPod(t, data, func(name string, nodeName string) error {
+	_, server0IPs, cleanupFunc := createAndWaitForPod(t, data, func(name string, nodeName string) error {
 		return data.createServerPod(name, "http", server0Port, false)
 	}, "test-server-", "")
 	defer cleanupFunc()
 
 	server1Port := 8080
-	_, server1IP, cleanupFunc := createAndWaitForPod(t, data, func(name string, nodeName string) error {
+	_, server1IPs, cleanupFunc := createAndWaitForPod(t, data, func(name string, nodeName string) error {
 		return data.createServerPod(name, "http", server1Port, false)
 	}, "test-server-", "")
 	defer cleanupFunc()
 
+	// TODO: add support for IPv6 after NetworkPolicy IPv6 is supported
+	if server0IPs.ipv4 == nil || server1IPs.ipv4 == nil {
+		t.Fatalf("No valid IPv4 address assigned for server Pod")
+	}
```

Comment:
this seems redundant with `skipIfNotIPv4Cluster` (same comment for other test cases)?

Reply:
They are not the same. `skipIfNotIPv4Cluster` is checking the cluster configurations include IPv4 Pod CIDR, which means both IPv4 only and dual-stack cluster could run the test cases. As function `podWaitForIPs` only checks that Pod is assigned at least one IP address when it is running, an error might exists that IPv4 is not assigned in a dual-stack cluster. So I add extra check in each case for the existence of IPv4 address. Maybe I should move the check to function `podWaitForIPs`, then we could consume the IPv4 address in the test cases directly if it has mark `skipIfNotIPv4Cluster`, what do you think @antoninbas ?

Reply:
Yes I think it would be a good idea to move the checks to `podWaitForIPs` to avoid repeating the same checks in multiple test cases.

Reply:
removed.